### PR TITLE
Review of AWS modules and subsequent correction of RKE/RKE2/K3S recipes

### PR DESCRIPTION
**Pull request by glovecchi0**
Reviewed AWS modules and rewrote AWS EC2 x RKE recipe.

No features have been modified or added, but only some issues have been resolved, including: #99 , #118 , #119 , and #120 .

The documentation has also been revised.

tfvars example:
``` bash
prefix           = "glovecchio"
aws_region       = "eu-west-1"
instance_count   = 2
ssh_username     = "ubuntu"
rancher_hostname = "glovecchio"
rancher_password = "Mypassword.1234"
```

### CODE REVIEW

**Comment by glovecchi0** [comment]
Rewrote AWS EC2 x RKE (split-role) recipe.

The documentation has also been revised.

tfvars example:
```bash
prefix             = "glovecchio"
aws_region         = "eu-west-1"
master_nodes_count = 1
worker_nodes_count = 1
ssh_username       = "ubuntu"
rancher_hostname   = "glovecchio"
rancher_password   = "Mypassword.1234"
```

**Comment by glovecchi0** [comment]
Rewrote AWS EC2 x RKE2 recipe.

The documentation has also been revised.

tfvars example:
```bash
prefix           = "glovecchio"
aws_region       = "eu-west-1"
instance_count   = 2
ssh_username     = "ubuntu"
rancher_hostname = "glovecchio"
rancher_password = "Mypassword.1234"
```

**Comment by glovecchi0** [comment]
I will fix conflicts at the end of the review.

**Comment by glovecchi0** [comment]
Added Rancher deployment to the AWS EC2 x RKE (split-role) recipe.

tfvars example:
```bash
prefix             = "glovecchio"
aws_region         = "eu-west-1"
server_nodes_count = 1
worker_nodes_count = 1
ssh_username       = "ubuntu"
rancher_hostname   = "glovecchio"
rancher_password   = "Mypassword.1234"
```

**Comment by glovecchi0** [comment]
Rewrote AWS EC2 x K3S recipe.

The documentation has also been revised.

tfvars example:
```bash
prefix             = "glovecchio"
aws_region         = "eu-west-1"
server_nodes_count = 1
worker_nodes_count = 1
ssh_username       = "ubuntu"
rancher_hostname   = "glovecchio"
rancher_password   = "Mypassword.1234"
```

**Review by ajithijk — APPROVED** [review]
Tested all and worked very well.

File: recipes/upstream/aws/k3s/README.md [hunk 1]
```diff
@@ -11,77 +11,30 @@ git clone https://github.com/rancherlabs/tf-rancher-up.git
...
+    -  `rancher_password` to configure the initial Admin password (the password must be at least 12 characters)
+- Make sure you are logged into your AWS Account from your local Terminal. See the preparatory steps [here](../../../../modules/infra/aws/README.md).
 
-- If storing multiple AWS credentials in `~/.aws/credentials`, set the profile when running terraform.
+**NB: If you want to use all the configurable variables in the `terraform.tfvars` file, you will need to uncomment them there and in the `variables.tf` and `main.tf` files.**
 
 ```bash
-AWS_PROFILE=<profile name> terraform plan -var-file terraform.tfvars
-AWS_PROFILE=<profile name> terraform apply -var-file terraform.tfvars
+terraform init -upgrade ; terraform apply -target=module.k3s-first-server.tls_private_key.ssh_private_key -target=module.k3s-first-server.local_file.private_key_pem -target=module.k3s-first-server.local_file.public_key_pem -target=module.k3s-first-server.aws_key_pair.key_pair -target=module.k3s-first-server.aws_vpc.vpc -target=module.k3s-first-server.aws_subnet.subnet -target=module.k3s-first-server.aws_security_group.sg_allowall -auto-approve ; terraform apply -auto-approve
```
Comment (leodotcloud) [inline recipes/upstream/aws/k3s/README.md:29]: Let's explore if there is a way to avoid such a long `terraform apply`.

**Review by leodotcloud — CHANGES_REQUESTED** [review]

File: recipes/upstream/aws/k3s/README.md (hunk 1 above)
Comment (glovecchi0) [reply recipes/upstream/aws/k3s/README.md]: @leodotcloud, fixed as requested; now everything works with a single command.

Comment (glovecchi0) [reply recipes/upstream/aws/k3s/README.md]: cc @ajithijk , @dnoland1

**Comment by leodotcloud** [comment]
@glovecchi0 can you take care of the conflicts?

**Comment by glovecchi0** [comment]
> @glovecchi0 can you take care of the conflicts?

Yes; I'll work on It.

**Review by leodotcloud — APPROVED** [review]

**Pull request closed without merging**